### PR TITLE
add the java max memory used flag.

### PR DESCRIPTION
**Pull request by sleongmgi**

### CODE REVIEW

**Comment by acoffman** [comment]
+1

File: lib/perl/Genome/Model/Tools/Varscan.pm [hunk 1]
```diff
@@ -66,14 +89,20 @@ my %VARSCAN_VERSIONS = (
...
-    my $path = $self->path_for_version($self->version);
-    my $headers = $self->no_headers ? "--no-headers 1" : "";
-    my $command_line = 'java -jar ' . $path . ' ' . $parameter_string . ' '.$headers;
+    my @java_cmd = ( $self->java_interpreter );
+
+    my $path = $self->path_for_version( $self->version );
+    if ( defined $self->max_memory ) {
+        push @java_cmd, sprintf( "-Xmx%dg", $self->max_memory );
+    }
+    push @java_cmd, "-Djava.io.tmpdir=" . $self->tmp_dir, '-jar' => $path;
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Varscan.pm]: `tmp_dir` is currently not being set anywhere.  Perhaps [this fallback from G/M/T/Gatk/Base.pm](https://github.com/genome/genome/blob/5bf928f396ba5921abc6aaf715ccf43353fdb2eb/lib/perl/Genome/Model/Tools/Gatk/Base.pm#L95-L98) would be useful?

**Comment by tmooney** [comment]
+1.

I'm not keen on `sub max_memory` being copy+pasted wholesale from `Genome::Model::Tools::Gatk::Base`.  If it ends up being needed in a third location, we'll definitely need to come up with a better solution.

**Comment by sleongmgi** [comment]
I agreed.

**Pull request merged**